document/core/exec/instructions.rst

```diff
@@ -182,6 +182,489 @@ Where the underlying operators are non-deterministic, because they may return on
+
+2. Pop the value :math:`\V128.\CONST~c_1` from the stack.
+
+3. Let :math:`c` be the result of computing :math:`\vunop_{\shape}(c_1)`.
+
+4. Push the value :math:`\V128.\CONST~c` to the stack.
+
+.. math::
+   \begin{array}{lcl@{\qquad}l}
+   (\V128\K{.}\CONST~c_1)~\V128\K{.}\vunop &\stepto& (\V128\K{.}\CONST~c)
+     & (\iff c = \vunop_{\shape}(c_1))
+   \end{array}
+
+
+.. _exec-simd-vbinnop:
+
+:math:`\shape\K{.}\vbinop`
+..........................
+
+1. Assert: due to :ref:`validation <valid-vbinop>`, a value of :ref:`value type <syntax-valtype>` |V128| is on the top of the stack.
+
+2. Pop the value :math:`\V128.\CONST~c_2` from the stack.
+
+4. Pop the value :math:`\V128.\CONST~c_1` from the stack.
+
+4. If :math:`\vbinop_{\shape}(c_1, c_2)` is defined:
+
+   a. Let :math:`c` be a possible result of computing :math:`\vbinop_{\shape}(c_1, c_2)`.
+
+   b. Push the value :math:`\V128.\CONST~c` to the stack.
+
+5. Else:
+
+   a. Trap.
+
+.. math::
+   \begin{array}{lcl@{\qquad}l}
+   (\V128\K{.}\CONST~c_1)~(\V128\K{.}\CONST~c_2)~\V128\K{.}\vbinop &\stepto& (\V128\K{.}\CONST~c)
+     & (\iff c \in \vbinop_{\shape}(c_1, c_2)) \\
+   (\V128\K{.}\CONST~c_1)~(\V128\K{.}\CONST~c_2)~\V128\K{.}\vbinop &\stepto& \TRAP
+     & (\iff \vbinop_{\shape}(c_1, c_2) = \{\})
+   \end{array}
+
+
+.. _exec-simd-vshiftop:
+
+:math:`t\K{x}N\K{.}\vshiftop`
+.............................
+
+1. Assert: due to :ref:`validation <valid-vshiftop>`, a value of :ref:`value type <syntax-valtype>` |I32| is on the top of the stack.
+
+2. Pop the value :math:`\I32.\CONST~s` from the stack.
+
+3. Assert: due to :ref:`validation <valid-vshiftop>`, a value of :ref:`value type <syntax-valtype>` |V128| is on the top of the stack.
+
+4. Pop the value :math:`\V128.\CONST~c_1` from the stack.
+
+5. Let :math:`i_1^N` be the sequence :math:`\simdto_{\shape}(c_1)`.
+
+7. Let :math:`c` be :math:`\simdof_{\shape}(\vshiftop_{t}(i_1^N, s^N))`.
+
+8. Push the value :math:`\V128.\CONST~c` to the stack.
+
+.. math::
+   \begin{array}{l}
+   \begin{array}{lcl@{\qquad}l}
+   (\V128\K{.}\CONST~c_1)~(\I32\K{.}\CONST~s)~\V128\K{.}\vshiftop &\stepto& (\V128\K{.}\CONST~c)
+   \end{array}
+   \\ \qquad
+     \begin{array}[t]{@{}r@{~}l@{}}
+     (\iff & i_1^N = \simdto_{\shape}(c_1) \\
+     \wedge & c = \simdof_{\shape}(\vshiftop_{t}(i_1^N, s^N)))
+     \end{array}
+   \end{array}
+
+
+.. _exec-simd-vtestop:
+
+.. _exec-simd-alltrue:
+
+:math:`\shape\K{.}\ALLTRUE`
+...........................
+
+1. Assert: due to :ref:`validation <valid-vtestop>`, a value of :ref:`value type <syntax-valtype>` |V128| is on the top of the stack.
+
+2. Pop the value :math:`\V128.\CONST~c_1` from the stack.
+
+3. Let :math:`n` be the integer :math:`\lanes(\shape)`.
+
+4. Let :math:`i_1^n` be the sequence :math:`\simdto_{\shape}(c_1)`
+
+5. Let :math:`i` be the result of computing :math:`\bool((i_1 != 0)^n)`.
```

Comment:
use `\neq` instead of `!=`

Reply:
Also I'm not sure how to understand the logic here -- we convert each value in the sequence to a true or false value, then apply bool to convert to a sequence of 0 and 1, right? When is it collapsed to a single integer?

Reply:
Ah I'm missing something in the definition of `bool` to say that it works on a sequence on integers as well (AND over all of them).